### PR TITLE
Wait for connect/disconnect to update asset

### PR DESCRIPTION
**Pull request by mattJsonar**

### CODE REVIEW

File: dsfhub/resource_common.go [hunk 1]
```diff
@@ -408,13 +411,64 @@ func contains(l []string, x string) bool {
...
+
+	return nil
+}
+
+func auditStateRefreshFunc(client Client, resourceType string, assetId string) retry.StateRefreshFunc {
+	return func() (any, string, error) {
+		var result *ResourceWrapper
+		var err error
+
+		switch resourceType {
```
Comment (PetalJsonar) [inline dsfhub/resource_common.go:448]: nice! much cleaner :)

File: dsfhub/resource_common_test.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package dsfhub
+
+import (
+	"fmt"
+	"regexp"
+)
+
+func testAccParseResourceAttributeReference(field string) string {
+	var expr string = `dsfhub_[A-Za-z0-9_-].+\.[A-Za-z0-9_-].+` //e.g. dsfhub_cloud_account.my-cloud-account, dsfhub_cloud_account.my-cloud-account.asset_id
+	var ret string
```
Comment (PetalJsonar) [inline dsfhub/resource_common_test.go]: I know this repo is full of them and we won't go around changing all of those, but when creating new variables maybe we could take a note from Uncle Bob and make their names more readable?

File: dsfhub/resource_data_source_test.go [hunk 3]
```diff
@@ -40,62 +233,186 @@ func TestAccDSFDataSource_basic(t *testing.T) {
...
 		if err == nil {
-			return fmt.Errorf("DSF Data Source %s still exists for gatewayId: %s", dsfDataSourceId, testGatewayId)
+			return fmt.Errorf("DSF Data Source %s still exists", assetId)
 		}
 	}
 	return nil
 }
+
+// Configs
+func testAccDSFDataSourceConfig_Basic(resourceName string, adminEmail string, assetId string, gatewayId string, serverHostName string, serverType string) string {
```
Comment (PetalJsonar) [inline dsfhub/resource_data_source_test.go]: it's not necessary now but when we eventually rewrite the provider, it'd be nice to either import these from a test data file and/or rename them something slightly more intuitive to indicate that they're just data, not a test of their own

Comment (mattJsonar) [reply dsfhub/resource_data_source_test.go]: I'm just following [terraform's naming convention](https://developer.hashicorp.com/terraform/plugin/sdkv2/testing/acceptance-tests/testcase) of lowercase `testAcc` for function used by acceptance tests and then `Config_<config description>` to indicate it is test data. Not opposed to changing it though if you have better suggestions.

File: dsfhub/resource_common_test.go (hunk 2 above)
Comment (mattJsonar) [reply dsfhub/resource_common_test.go]: Updated the variable names and added a doc comment for the function

File: dsfhub/resource_data_source_test.go (hunk 3 above)
Comment (mattJsonar) [reply dsfhub/resource_data_source_test.go]: Liked the idea of having these separated out into their own files though, will make that change.

**Review by PetalJsonar — APPROVED** [review]

**Pull request merged**